alot/commands/thread.py

```diff
@@ -934,72 +934,130 @@ async def apply(self, ui):
                     raise CommandCanceled()
 
 
+@registerCommand(MODE, 'open', arguments=[
+    (['cmd'], {
+        'help': '''Shell command to use to open the attachment. \
+                   The path to the attachment file will be passed as an argument. \
+                   If absent, mailcap is used to select the command \
+                   based on the attachment's MIME type.''',
+        'nargs': '*',
+    }),
+    (['--thread'], {
+        'action': 'store_true',
+        'help': 'run in separate thread',
+    }),
+    (['--spawn'], {
+        'action': 'store_true',
+        'help': 'run in a new terminal window',
+    }),
+])
 class OpenAttachmentCommand(Command):
+    """opens an attachment with a given shell command
+       or according to mailcap"""
 
-    """displays an attachment according to mailcap"""
-    def __init__(self, attachment, **kwargs):
+    def __init__(self, cmd=(), thread=False, spawn=False, **kwargs):
         """
-        :param attachment: attachment to open
-        :type attachment: :class:`~alot.db.attachment.Attachment`
+        :param cmd: shell command to use to open the attachment
+        :type cmd: list of str
+        :param thread: whether to run in a separate thread
+        :type thread: bool
+        :param spawn: whether to run in a new terminal window
+        :type spawn: bool
         """
         Command.__init__(self, **kwargs)
-        self.attachment = attachment
+        self.cmd = cmd
+        self.thread = thread
+        self.spawn = spawn
 
     async def apply(self, ui):
-        logging.info('open attachment')
-        mimetype = self.attachment.get_content_type()
+        try:
+            logging.info('open attachment')
+            attachment = OpenAttachmentCommand._get_attachment(ui)
```

Comment:
isn't this the `OpenAttachmentCommand` class? why not simply call `_get_attachment`?

Reply:
Python wants the class name for static methods even within the class itself.

Reply:
aha. didn't know that. thanks.

Reply:
Amended.